.gitignore

```diff
@@ -21,3 +21,4 @@ e2e-tests
 concurrent_workspace_script_cache.json
 
 testDir
+**/amplify/*_env.ts
```

Comment:
Do we need this in our gitignore? Shouldn't the gitignore of `test-projects/*` cover this?

Reply:
Unit tests also cause the typedef generation to happen, for example in `backend-function/src/test-assets/default-lambda`. Not sure if we wanted those to be committed or not.

Reply:
We should make sure the unit tests clean up after themselves. But yeah it's okay to have the rule just so that if the unit tests fail, we don't end up committing test output

Reply:
If unit tests are localized in `backend-function` should we localize this expression too ?

Reply:
There is also a unit test generating typedefs in `integration-tests` as well, if I remove this in gitignore here are the untracked files:
```
packages/backend-function/src/test-assets/default-lambda/amplify/
packages/backend-function/src/test-assets/lambda-with-dependencies/amplify/
packages/integration-tests/src/test-projects/data-storage-auth-with-triggers-ts/amplify/data/echo/amplify/
packages/integration-tests/src/test-projects/data-storage-auth-with-triggers-ts/amplify/func-src/amplify/
packages/integration-tests/src/test-projects/standalone-data-auth-modes/amplify/data/amplify/
```

The problem with the unit test in `integration-tests` is that we can't mock `fs` so we would need to mock `FunctionEnvironmentTypeGenerator` which would add a dependency

Reply:
how about `**/test-assets/**/amplify` and `**/test-projects/**/amplify`(or so) ?
this at least scopes it down to be sure it's test resources.